core/math/vector3i.cpp

```diff
@@ -58,3 +58,53 @@ Vector3i Vector3i::clamp(const Vector3i &p_min, const Vector3i &p_max) const {
 Vector3i::operator String() const {
 	return "(" + itos(x) + ", " + itos(y) + ", " + itos(z) + ")";
 }
+
+Vector3i Vector3i::get_xyz() const {
+	return *this;
+}
+
+Vector3i Vector3i::get_xzy() const {
+	return Vector3i(x, z, y);
+}
+
+Vector3i Vector3i::get_yxz() const {
+	return Vector3i(y, x, z);
+}
+
+Vector3i Vector3i::get_yzx() const {
+	return Vector3i(y, z, x);
+}
+
+Vector3i Vector3i::get_zxy() const {
+	return Vector3i(z, x, y);
+}
+
+Vector3i Vector3i::get_zyx() const {
+	return Vector3i(z, y, x);
+}
+
+/**
```

Comment:
There is no such object as a `Vector2i` so I have commented out the related functions. Implementing `Vector2i` is outside of the scope of this PR, however I believe it to be valuable to leave these here for when that happens.

Reply:
There is `Vector2i`: https://github.com/godotengine/godot/blob/master/core/math/vector2.h#L280-L384

Reply:
Oh I see, it is part of the `Vector2` header. Should it be split into it's own file, much like how `Vector3i` is?